src/main/scala/scalismo/mesh/boundingSpheres/BSDistance.scala

```diff
@@ -22,7 +22,23 @@ import scalismo.mesh.boundingSpheres.ClosestPointType._
 /**
  * Holds triangles and precalculated vectors.
  */
-private[mesh] case class Triangle(a: EuclideanVector[_3D], b: EuclideanVector[_3D], c: EuclideanVector[_3D], ab: EuclideanVector[_3D], ac: EuclideanVector[_3D], n: EuclideanVector[_3D])
+private[mesh] case class Triangle(a: EuclideanVector[_3D], b: EuclideanVector[_3D], c: EuclideanVector[_3D]) {
+  val ab = b - a
```

Comment:
These could be made private (though it does not matter much as the class is private and used only locally)